code/messaging/src/main/java/com/adobe/marketing/mobile/MessagingPushUtils.java

```diff
@@ -0,0 +1,85 @@
+/*
+  Copyright 2023 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+
+package com.adobe.marketing.mobile;
+
+import android.content.ContentResolver;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.Uri;
+import androidx.annotation.NonNull;
+import com.adobe.marketing.mobile.services.Log;
+import com.adobe.marketing.mobile.util.StringUtils;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+class MessagingPushUtils {
+    private static final String SELF_TAG = "PushNotificationMediaDownloader";
+    private final Context context;
```

Comment:
Recommend using the `context` as a parmeter to the utility function where required instead of holding a strong reference to Context here. The caller can fetch it via `ServiceProvider.getInstance.getApplication().getApplicationContex()`.  That will allow you to make this a true singleton and the methods static.